app/models/discussion.rb

```diff
@@ -8,10 +8,6 @@ class Discussion < ActiveRecord::Base
 
   default_scope -> { order('updated_at DESC') }
 
-  after_save :update_subscriptions
```

Comment:
Actually there's no problem here. Even if the callback gets triggered twice in a transaction, the subscription won't change states again, and no new notification will be generated